index.d.ts

```diff
@@ -0,0 +1,128 @@
+declare module 'fluture' {
+  interface ResolveFunction<T> {
+    (value: T): void
+  }
+  interface RejectFunction<S> {
+    (error: S): void
+  }
+
+  interface FunctionReturning<T> {
+    (...t: any[]): T
+  }
+  abstract class Future<T, S> {
```

Comment:
I would prefer that the positions of the generic types correspond to the positions of the functions in `fork`. In my example I named them `L` and `R` to mean "left" and "right" respectively. We can keep the names `S` and `T`, if that's a convention of sorts, but I would prefer the generic on the left to represent the type of the *rejection* reason, and the generic on the right to represent the *resolution* value.